pkg/operator/controller/ingress/status.go

```diff
@@ -943,12 +955,38 @@ func computeLoadBalancerProgressingStatus(ic *operatorv1.IngressController, serv
 	}
 }
 
+// clearIngressControllerInactiveAWSLBTypeParametersStatus clears AWS parameters for the inactive load balancer type,
+// unless there is a load balancer type change is pending. When a change is pending, setDefaultPublishingStrategy
+// sets the desired parameters for the new LB type. To avoid removing the desired state, we allow
```

Comment:
The logic of  `setDefaultPublishingStategy` puzzled me in other reviews too. Do I understand correctly that this function reverts the changes done by `setDefaultPublishingStategy` syncing the status with the actual
 state of the service (good direction)?

Reply:
Not for this PR but why can't we just stop updating the status in `setDefaultPublishingStategy` and let `syncIngressControllerStatus` take care of the status exclusively?
 
 If the type of the LB is changed by the user:
- we save the desired state in `spec`
- update the status to instruct the user to recreate the service
- keep the LB type in `status` the same as in the service's annotation

Once the service is manually recreated the reconciliation will be triggered and we will reset the status to the new type taken from the service annotation.

Am I missing some hidden value in what `setDefaultPublishingStategy` is doing w.r.t. to the status?

Reply:
Right, I originally did this status NLB/CLB parameters clearing in `setDefaultPublishingStategy`, then I noticed during pending Scope changes that `setDefaultPublishingStategy` was only keeping the desired parameters and deleting the actual parameter's status (it doesn't know or care about pending changes, it just know and cares about desired state).

So, we can't just clear the NLB/CLB parameters based on desired state, we need to know the actual state (if a change is pending), so we don't prematurely clear status that represents the actual state.

> Not for this PR but why can't we just stop updating the status in setDefaultPublishingStategy and let syncIngressControllerStatus take care of the status exclusively?

Most of the time yes. The only exception is maybe some defaulting logic, especially in `setDefaultProviderParameters` that may need to be added to or maintained. I wanted to put a comment in `setDefaultPublishingStategy` to prevent future developers from adding logic to it accidentally (like what happened with Floating IP). I might go ahead and do that.

> If the type of the LB is changed by the user:
> we save the desired state in spec
> update the status to instruct the user to recreate the service
> keep the LB type in status the same as in the service's annotation

Yes, in a perfect world, but to be clear, this PR doesn't `keep the LB type in status the same as in the service's annotation`, it actually immediately updates the LB type in the status to the desired state while it's pending. Hongan was asking the same [question](https://github.com/openshift/cluster-ingress-operator/pull/1142#issuecomment-2345315070). API Team said that changing the behavior of a status field is incompatible, and we also use the LB Type status field for defaulting, which cause us to loose our defaulting. So this was a dead end.

> Am I missing some hidden value in what setDefaultPublishingStategy is doing w.r.t. to the status?

Not really, but maybe some in memory defaulting to avoid nil pointer exceptions like I mentioned earlier.